File: app/src/main/java/org/wen/gank/widgets/DividerItemDecoration.java
package org.wen.gank.widgets;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Rect;
import android.support.annotation.ColorInt;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * created by Jiahui.wen 2017-09-20
 */
public class DividerItemDecoration extends RecyclerView.ItemDecoration {

    private final int dividerHeight;
    private final Paint paint;

    public DividerItemDecoration(@ColorInt int dividerColor, int dividerHeight) {
        this.dividerHeight = dividerHeight;
        paint = new Paint(Paint.ANTI_ALIAS_FLAG);
        paint.setColor(dividerColor);
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent,
        RecyclerView.State state) {
        outRect.bottom += dividerHeight;
    }

    @Override
    public void onDrawOver(Canvas c, RecyclerView parent, RecyclerView.State state) {
        super.onDrawOver(c, parent, state);

        int childCount = parent.getChildCount();
        int left = parent.getPaddingLeft();
        int right = parent.getWidth() - parent.getPaddingRight();
        for (int i = 0; i < childCount; i++) {
            View child = parent.getChildAt(i);

            int top = child.getBottom();
            int bottom = top + dividerHeight;
            c.drawRect(left, top, right, bottom, paint);
        }
    }
}

File: app/src/main/java/org/wen/gank/widgets/DividerItemDecoration.kt
package org.wen.gank.widgets

import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Rect
import android.view.View
import androidx.annotation.ColorInt
import androidx.recyclerview.widget.RecyclerView

/**
 * created by Jiahui.wen 2017-09-20
 */
class DividerItemDecoration(@ColorInt dividerColor: Int, private val dividerHeight: Int) : RecyclerView.ItemDecoration() {
  private val paint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)

  init {
    paint.color = dividerColor
  }

  override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView,
                              state: RecyclerView.State) {
    outRect.bottom += dividerHeight
  }

  override fun onDrawOver(c: Canvas, parent: RecyclerView, state: RecyclerView.State) {
    super.onDrawOver(c, parent, state)

    val childCount = parent.childCount
    val left = parent.paddingLeft
    val right = parent.width - parent.paddingRight
    for (i in 0 until childCount) {
      val child = parent.getChildAt(i)

      val top = child.bottom
      val bottom = top + dividerHeight
      c.drawRect(left.toFloat(), top.toFloat(), right.toFloat(), bottom.toFloat(), paint)
    }
  }
}
